### PR TITLE
Injectable Button Config #15989

### PR DESCRIPTION
**Pull request by jonathanRinciari**

### CODE REVIEW

**Comment by jonathanRinciari** [comment]
just figured out how to actually update the api files!

File: src/material/button/button.ts [hunk 1]
```diff
@@ -35,6 +36,27 @@ import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
 /** Default color palette for round buttons (mat-fab and mat-mini-fab) */
 const DEFAULT_ROUND_BUTTON_COLOR = 'accent';
 
+/* Default Button Type that can be overridden*/
```
Comment (mkonars) [inline src/material/button/button.ts]: Should it be /* Default Button options that can be overriden */ ?

File: src/material/button/button.ts [hunk 2]
```diff
@@ -35,6 +36,27 @@ import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
...
+export interface MatButtonDefaultOptions {
+  /** Default Button Type */
+  type: 'submit' | 'reset' | 'button';
+
+  [key: string]: string;
```
Comment (mkonars) [inline src/material/button/button.ts]: Hi, sorry if this is a stupid question, but is this needed?

File: src/material/button/button.ts [hunk 3]
```diff
@@ -91,9 +113,16 @@ export class MatButton extends _MatButtonMixinBase
 
   constructor(elementRef: ElementRef,
               private _focusMonitor: FocusMonitor,
+              @Inject(MAT_BUTTON_DEFAULT_OPTIONS) defaults: MatButtonDefaultOptions,
               @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode: string) {
     super(elementRef);
 
+    for (const key in defaults) {
+      // Allow user to override default options
+      if (!this._hasHostAttributes('type')) {
```
Comment (mkonars) [inline src/material/button/button.ts]: Should it rather be: ._hasHostAttributes(attr)) ?

**Comment by andrewseguin** [comment]
Missing PR description and inactivity after review was left - if this is something you'd still like to contribute, feel free to reopen a rebased version

**Pull request closed without merging**